crates/sui-core/src/consensus_adapter.rs

```diff
@@ -106,6 +108,10 @@ pub struct ConsensusAdapter {
     num_inflight_transactions: AtomicU64,
     /// A structure to register metrics
     opt_metrics: OptArcConsensusAdapterMetrics,
+    /// Pending certificates that we are waiting to be sequenced by consensus
+    /// Lock ordering: this is a 'leaf' lock, no other locks should be acquired in the scope of this lock
+    /// In particular, this lock is always acquired after taking read or write lock on reconfig state
```

Comment:
By "after", you don't mean while we are hooding the reconfig state lock, right?

Reply:
I do mean that (and that is what we do sometimes - holding two locks together).